### PR TITLE
used an api to get posts and then render them on the DOM 

### PR DESCRIPTION
**Pull request by ChoudharyHamza**
used axios to make the pull request and get an array of 100 objects each containing id, userid, title, body and then rendred them on the dom using a component GivePosts

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -1,6 +1,9 @@
 import Vue from 'vue'
 import App from './App.vue'
 
+
+
+
```
Comment (ayousaf11) [inline src/main.js]: @ChoudharyHamza extra blank lines here

File: src/components/givePost.vue [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+</script>
+
+<style >
+.container{
+  width:90%;
+  margin-left: 5%;
+  margin-right:5%;
+  margin-top:50px;
+}
+#main-heading{
```
Comment (ayousaf11) [inline src/components/givePost.vue]: @ChoudharyHamza `class` is proffered for styles unless needed

File: src/App.vue [hunk 3]
```diff
@@ -1,28 +1,37 @@
...
+/* eslint-disable */
+import axios from "axios"
+import givePost from "./components/givePost"
 export default {
   name: 'app',
-  components: {
-    HelloWorld
-  }
+  components:{
+    givePost
```
Comment (ayousaf11) [inline src/App.vue]: @ChoudharyHamza make meaningful component names.

File: src/main.js (hunk 1 above)
Comment (ChoudharyHamza) [reply src/main.js]: ok have removed the spaces and will avoid them

File: src/App.vue (hunk 3 above)
Comment (ChoudharyHamza) [reply src/App.vue]: ok will keeo in mind and have changed the name from givePost to RenderPosts

File: src/components/givePost.vue (hunk 2 above)
Comment (ChoudharyHamza) [reply src/components/givePost.vue]: ok used class from this point on and removed the id's used on previous elements

File: src/App.vue (hunk 3 above)
Comment (ChoudharyHamza) [reply src/App.vue]: also added the pagination feature.

File: src/components/postPagination.vue [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    <div class="btn-con">
+        {{this.buttonsRen()}}
+    <button class="btn" v-for="button in buttons" @click="paginationAlgo(button)" :class="currentPage(button)" :key="button">{{ button }}</button>
+        </div>
+</template>
+
+<script>
+/*eslint-disable*/
+export default {
+    props:["posts","paginationAlgo", "activePage"],
```
Comment (ayousaf11) [inline src/components/postPagination.vue]: @ChoudharyHamza make meaningful variable names. `posts` prop does not represent the posts but count.

Comment (ChoudharyHamza) [reply src/components/postPagination.vue]: ok will be carefull from now on

Comment (ayousaf11) [reply src/components/postPagination.vue]: @ChoudharyHamza once you fix the name I will approve the merge and you will be good to merge.

Comment (ChoudharyHamza) [reply src/components/postPagination.vue]: ok made the changes to the project

**Comment by ayousaf11** [comment]
@ChoudharyHamza once you merge the branch in `master`, rebase your 2nd branch from the `master`. In this way you will be able to get the latest of `master` in the branch you are working on. Thanks!

**Pull request merged**